From ebedd9f0cc633adc5fa80247306835b376467b9c Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Fri, 27 Oct 2023 10:08:31 +0200
Subject: [PATCH] Add data files for themes and plugins using setuptools
 package-data

Use setuptools' `package-data` [1] to unconditionally add all files
below sigal's `themes/` dir and all data files below the
`plugins/encrypt/static/` dir to a wheel.
This works without any SCM integration and without `MANIFEST.in`.

Fixes https://github.com/saimn/sigal/issues/500

[1] https://setuptools.pypa.io/en/latest/userguide/datafiles.html#package-data

Signed-off-by: David Runge <dave@sleepmap.de>
---
 pyproject.toml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/pyproject.toml b/pyproject.toml
index 7ed1575..df3bbbd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,6 +55,12 @@ license-files = ["LICENSE"]
 where = ["src"]
 namespaces = true
 
+[tool.setuptools.package-data]
+sigal = [
+    "plugins/encrypt/static/*",
+    "themes/**",
+]
+
 [tool.setuptools_scm]
 write_to = "src/sigal/version.py"
 
-- 
2.42.0

